iceoryx_posh/test/moduletests/test_roudi_portmanager.cpp

```diff
@@ -446,29 +439,105 @@ TEST_F(PortManager_test, ConditionVariablesDestroy)
     }
 }
 
+TEST_F(PortManager_test, NodeDataOverflow)
+{
+    std::string process = "Process";
+    std::string node = "Node";
+
+    // first acquire all possible NodeData
+    for (unsigned int i = 0U; i < iox::MAX_NODE_NUMBER; i++)
+    {
+        iox::ProcessName_t newProcessName(iox::cxx::TruncateToCapacity, process + std::to_string(i));
+        iox::NodeName_t newNodeName(iox::cxx::TruncateToCapacity, node + std::to_string(i));
+        auto nodeData = m_portManager->acquireNodeData(newProcessName, newNodeName);
+        EXPECT_THAT(nodeData.has_error(), Eq(false));
+    }
+
+    // test if overflow errors get hit
+    {
+        auto errorHandlerCalled{false};
+        auto errorHandlerGuard = iox::ErrorHandler::SetTemporaryErrorHandler(
+            [&errorHandlerCalled](const iox::Error, const std::function<void()>, const iox::ErrorLevel) {
+                errorHandlerCalled = true;
+            });
+
+        auto nodeData = m_portManager->acquireNodeData("AnotherProcess", "AnotherNode");
+        EXPECT_THAT(nodeData.has_error(), Eq(true));
+        EXPECT_THAT(errorHandlerCalled, Eq(true));
+        EXPECT_THAT(nodeData.get_error(), Eq(PortPoolError::NODE_DATA_LIST_FULL));
+    }
+
+    // delete one and add one NodeData should be possible now
+    {
+        unsigned int testi = 0U;
+        iox::ProcessName_t newProcessName(iox::cxx::TruncateToCapacity, process + std::to_string(testi));
+        iox::NodeName_t newNodeName(iox::cxx::TruncateToCapacity, node + std::to_string(testi));
+        m_portManager->deletePortsOfProcess(newProcessName);
+
+        auto nodeData = m_portManager->acquireNodeData(newProcessName, newNodeName);
+        EXPECT_THAT(nodeData.has_error(), Eq(false));
+        EXPECT_THAT(nodeData.value()->m_node, StrEq(newNodeName));
+    }
+}
+
+TEST_F(PortManager_test, NodeDataDestroy)
```

Comment:
The purpose of this test seems to be to show that destroyed node data enables us to recycle the memory to create new node data. The test name could be improved to reflect this.